### PR TITLE
Furi: core refactoring and CMSIS removal part 2 

### PR DESCRIPTION
**Pull request by skotopes**
# What's new

- Furi: core refactoring and CMSIS removal part 2 

# Verification 

- Compile and upload
- Do release and unit tests 

# Checklist (For Reviewer)

- [x] PR has description of feature/bug or link to Confluence/Jira task
- [x] Description contains actions to verify feature/bugfix
- [x] I've built this code, uploaded it to the device and verified feature/bugfix

### CODE REVIEW

File: applications/dialogs/view_holder.c [hunk 1]
```diff
@@ -96,7 +96,7 @@ void view_holder_start(ViewHolder* view_holder) {
 }
 
 void view_holder_stop(ViewHolder* view_holder) {
-    while(view_holder->ongoing_input) osDelay(1);
+    while(view_holder->ongoing_input) furi_delay_ms(1);
```
Comment (DrZlo13) [inline applications/dialogs/view_holder.c]: furi_delay_tick

File: applications/input/input.c [hunk 2]
```diff
@@ -127,9 +127,9 @@ int32_t input_srv() {
         }
 
         if(is_changing) {
-            osDelay(1);
+            furi_delay_ms(1);
```
Comment (DrZlo13) [inline applications/input/input.c]: furi_delay_tick

File: applications/music_player/music_player_worker.c [hunk 3]
```diff
@@ -78,10 +78,10 @@ static int32_t music_player_worker_thread_callback(void* context) {
 
             furi_hal_speaker_stop();
             furi_hal_speaker_start(frequency, volume);
-            while(instance->should_work && furi_hal_get_tick() < next_tick) {
+            while(instance->should_work && furi_get_tick() < next_tick) {
                 volume *= 0.9945679;
                 furi_hal_speaker_set_volume(volume);
-                furi_hal_delay_ms(2);
+                furi_delay_ms(2);
```
Comment (DrZlo13) [inline applications/music_player/music_player_worker.c:84]: furi_delay_tick

File: firmware/targets/f7/ble_glue/ble_glue.c [hunk 4]
```diff
@@ -201,7 +201,7 @@ bool ble_glue_wait_for_c2_start(int32_t timeout) {
         started = ble_glue->status == BleGlueStatusC2Started;
         if(!started) {
             timeout--;
-            osDelay(1);
+            furi_delay_ms(1);
```
Comment (DrZlo13) [inline firmware/targets/f7/ble_glue/ble_glue.c]: furi_delay_tick

File: applications/unit_tests/rpc/rpc_test.c [hunk 5]
```diff
@@ -87,7 +86,7 @@ static void test_rpc_setup(void) {
     rpc = furi_record_open("rpc");
     for(int i = 0; !(rpc_session[0].session) && (i < 10000); ++i) {
         rpc_session[0].session = rpc_session_open(rpc);
-        furi_hal_delay_ms(1);
+        furi_delay_ms(1);
```
Comment (DrZlo13) [inline applications/unit_tests/rpc/rpc_test.c]: furi_delay_tick

File: applications/unit_tests/rpc/rpc_test.c [hunk 6]
```diff
@@ -107,7 +106,7 @@ static void test_rpc_setup_second_session(void) {
 
     for(int i = 0; !(rpc_session[1].session) && (i < 10000); ++i) {
         rpc_session[1].session = rpc_session_open(rpc);
-        furi_hal_delay_ms(1);
+        furi_delay_ms(1);
```
Comment (DrZlo13) [inline applications/unit_tests/rpc/rpc_test.c]: furi_delay_tick

File: applications/subghz/views/receiver.c [hunk 7]
```diff
@@ -72,7 +72,7 @@ void subghz_view_receiver_set_lock(SubGhzViewReceiver* subghz_receiver, SubGhzLo
                 model->bar_show = SubGhzViewReceiverBarShowLock;
                 return true;
             });
-        osTimerStart(subghz_receiver->timer, pdMS_TO_TICKS(1000));
+        furi_timer_start(subghz_receiver->timer, pdMS_TO_TICKS(1000));
```
Comment (DrZlo13) [inline applications/subghz/views/receiver.c:75]: FURI_MS_TO_TICKS ?

File: furi/furi.c [hunk 8]
```diff
@@ -0,0 +1,31 @@
+#include "furi.h"
+#include <string.h>
+#include "queue.h"
+
+void furi_init() {
+    furi_assert(!furi_is_irq_context());
+    furi_assert(xTaskGetSchedulerState() == taskSCHEDULER_NOT_STARTED);
+
+    CoreDebug->DEMCR |= CoreDebug_DEMCR_TRCENA_Msk;
```
Comment (DrZlo13) [inline furi/furi.c]: Platform specific code in the core?

File: lib/subghz/subghz_tx_rx_worker.c [hunk 9]
```diff
@@ -68,11 +68,11 @@ bool subghz_tx_rx_worker_rx(SubGhzTxRxWorker* instance, uint8_t* data, uint8_t*
     if(instance->status != SubGhzTxRxWorkerStatusRx) {
         furi_hal_subghz_rx();
         instance->status = SubGhzTxRxWorkerStatusRx;
-        osDelay(1);
+        furi_delay_ms(1);
```
Comment (DrZlo13) [inline lib/subghz/subghz_tx_rx_worker.c]: furi_delay_tick

File: lib/subghz/subghz_tx_rx_worker.c [hunk 10]
```diff
@@ -68,11 +68,11 @@ bool subghz_tx_rx_worker_rx(SubGhzTxRxWorker* instance, uint8_t* data, uint8_t*
...
     }
     //waiting for reception to complete
     while(furi_hal_gpio_read(&gpio_cc1101_g0)) {
-        osDelay(1);
+        furi_delay_ms(1);
```
Comment (DrZlo13) [inline lib/subghz/subghz_tx_rx_worker.c]: furi_delay_tick

File: lib/subghz/subghz_tx_rx_worker.c [hunk 11]
```diff
@@ -106,14 +106,14 @@ void subghz_tx_rx_worker_tx(SubGhzTxRxWorker* instance, uint8_t* data, size_t si
     furi_hal_subghz_tx(); //start send
     instance->status = SubGhzTxRxWorkerStatusTx;
     while(!furi_hal_gpio_read(&gpio_cc1101_g0)) { // Wait for GDO0 to be set -> sync transmitted
-        osDelay(1);
+        furi_delay_ms(1);
```
Comment (DrZlo13) [inline lib/subghz/subghz_tx_rx_worker.c]: furi_delay_tick

File: lib/subghz/subghz_tx_rx_worker.c [hunk 12]
```diff
@@ -106,14 +106,14 @@ void subghz_tx_rx_worker_tx(SubGhzTxRxWorker* instance, uint8_t* data, size_t si
...
         if(!--timeout) {
             FURI_LOG_W(TAG, "TX !cc1101_g0 timeout");
             break;
         }
     }
     while(furi_hal_gpio_read(&gpio_cc1101_g0)) { // Wait for GDO0 to be cleared -> end of packet
-        osDelay(1);
+        furi_delay_ms(1);
```
Comment (DrZlo13) [inline lib/subghz/subghz_tx_rx_worker.c]: furi_delay_tick

File: lib/subghz/subghz_tx_rx_worker.c [hunk 13]
```diff
@@ -189,7 +189,7 @@ static int32_t subghz_tx_rx_worker_thread(void* context) {
         }
 
         if(timeout_tx) timeout_tx--;
-        osDelay(1);
+        furi_delay_ms(1);
```
Comment (DrZlo13) [inline lib/subghz/subghz_tx_rx_worker.c]: furi_delay_tick

**Review by DrZlo13 — CHANGES_REQUESTED** [review]

File: applications/desktop/animations/animation_manager.c [hunk 14]
```diff
@@ -198,7 +198,7 @@ static void animation_manager_start_new_idle(AnimationManager* animation_manager
     const BubbleAnimation* bubble_animation =
         animation_storage_get_bubble_animation(animation_manager->current_animation);
     animation_manager->state = AnimationManagerStateIdle;
-    osTimerStart(animation_manager->idle_animation_timer, bubble_animation->duration * 1000);
+    furi_timer_start(animation_manager->idle_animation_timer, bubble_animation->duration * 1000);
```
Comment (hedger) [inline applications/desktop/animations/animation_manager.c:201]: furi_ms_to_ticks ?

File: applications/desktop/animations/animation_manager.c [hunk 15]
```diff
@@ -504,13 +504,13 @@ void animation_manager_load_and_continue_animation(AnimationManager* animation_m
...
-                        osTimerStart(
+                        furi_timer_start(
                             animation_manager->idle_animation_timer,
                             animation_manager->freezed_animation_time_left);
                     } else {
                         const BubbleAnimation* animation = animation_storage_get_bubble_animation(
                             animation_manager->current_animation);
-                        osTimerStart(
+                        furi_timer_start(
                             animation_manager->idle_animation_timer, animation->duration * 1000);
```
Comment (hedger) [inline applications/desktop/animations/animation_manager.c:514]: furi_ms_to_ticks?

File: applications/desktop/animations/animation_manager.c [hunk 16]
```diff
@@ -504,13 +504,13 @@ void animation_manager_load_and_continue_animation(AnimationManager* animation_m
                     animation_manager->state = AnimationManagerStateIdle;
 
                     if(animation_manager->freezed_animation_time_left) {
-                        osTimerStart(
+                        furi_timer_start(
                             animation_manager->idle_animation_timer,
                             animation_manager->freezed_animation_time_left);
```
Comment (hedger) [inline applications/desktop/animations/animation_manager.c:509]: furi_ms_to_ticks?

File: applications/gui/icon_animation.c [hunk 17]
```diff
@@ -7,15 +7,16 @@ IconAnimation* icon_animation_alloc(const Icon* icon) {
...
+        furi_timer_alloc(icon_animation_timer_callback, FuriTimerTypePeriodic, instance);
     return instance;
 }
 
 void icon_animation_free(IconAnimation* instance) {
     furi_assert(instance);
     icon_animation_stop(instance);
-    while(xTimerIsTimerActive(instance->timer) == pdTRUE) osDelay(1);
-    furi_check(osTimerDelete(instance->timer) == osOK);
+    while(xTimerIsTimerActive(instance->timer) == pdTRUE) furi_delay_ms(1);
```
Comment (hedger) [inline applications/gui/icon_animation.c]: make xTimerIsTimerActive private? + os_delay_ticks

File: applications/gui/icon_animation.c [hunk 18]
```diff
@@ -68,7 +69,7 @@ void icon_animation_start(IconAnimation* instance) {
         furi_check(
             xTimerChangePeriod(
```
Comment (hedger) [inline applications/gui/icon_animation.c:70]: make xTimerChangePeriod private?

File: applications/gui/modules/popup.c [hunk 19]
```diff
@@ -93,24 +93,25 @@ static bool popup_view_input_callback(InputEvent* event, void* context) {
 void popup_start_timer(void* context) {
     Popup* popup = context;
     if(popup->timer_enabled) {
-        uint32_t timer_period = popup->timer_period_in_ms / (1000.0f / osKernelGetTickFreq());
+        uint32_t timer_period =
```
Comment (hedger) [inline applications/gui/modules/popup.c:96]: furi_ms_to_ticks?

File: applications/gui/modules/text_input.c [hunk 20]
```diff
@@ -310,7 +310,7 @@ static void text_input_handle_ok(TextInput* text_input, TextInputModel* model, b
            (!model->validator_callback(
                model->text_buffer, model->validator_text, model->validator_callback_context))) {
             model->valadator_message_visible = true;
-            osTimerStart(text_input->timer, osKernelGetTickFreq() * 4);
+            furi_timer_start(text_input->timer, furi_kernel_get_tick_frequency() * 4);
```
Comment (hedger) [inline applications/gui/modules/text_input.c:313]: furi_ms_to_ticks ?

File: applications/input/input.c [hunk 21]
```diff
@@ -4,18 +4,18 @@
 
 static Input* input = NULL;
 
-inline static void input_timer_start(osTimerId_t timer_id, uint32_t ticks) {
+inline static void input_timer_start(FuriTimer* timer_id, uint32_t ticks) {
     TimerHandle_t hTimer = (TimerHandle_t)timer_id;
     furi_check(xTimerChangePeriod(hTimer, ticks, portMAX_DELAY) == pdPASS);
```
Comment (hedger) [inline applications/input/input.c:9]: make xTimerChangePeriod private?

File: applications/input/input.c [hunk 22]
```diff
@@ -4,18 +4,18 @@
...
 
-inline static void input_timer_stop(osTimerId_t timer_id) {
+inline static void input_timer_stop(FuriTimer* timer_id) {
     TimerHandle_t hTimer = (TimerHandle_t)timer_id;
     furi_check(xTimerStop(hTimer, portMAX_DELAY) == pdPASS);
     // xTimerStop is not actually stopping timer,
     // Instead it places stop event into timer queue
     // This code ensures that timer is stopped
-    while(xTimerIsTimerActive(hTimer) == pdTRUE) osDelay(1);
+    while(xTimerIsTimerActive(hTimer) == pdTRUE) furi_delay_ms(1);
```
Comment (hedger) [inline applications/input/input.c]: furi_delay_ticks + make xTimerIsTimerActive private?

File: applications/lfrfid/helpers/rfid_reader.cpp [hunk 23]
```diff
@@ -39,12 +39,12 @@ void RfidReader::decode(bool polarity) {
 }
 
 bool RfidReader::switch_timer_elapsed() {
-    const uint32_t seconds_to_switch = osKernelGetTickFreq() * 2.0f;
-    return (osKernelGetTickCount() - switch_os_tick_last) > seconds_to_switch;
+    const uint32_t seconds_to_switch = furi_kernel_get_tick_frequency() * 2.0f;
```
Comment (hedger) [inline applications/lfrfid/helpers/rfid_reader.cpp:42]: use ticks/ms coverter? @DrZlo13

File: applications/music_player/music_player_worker.c [hunk 24]
```diff
@@ -51,20 +51,20 @@ static int32_t music_player_worker_thread_callback(void* context) {
     while(instance->should_work) {
         if(NoteBlockArray_end_p(it)) {
             NoteBlockArray_it(it, instance->notes);
-            osDelay(10);
+            furi_delay_ms(10);
```
Comment (hedger) [inline applications/music_player/music_player_worker.c:54]: _delay_ticks?

File: applications/nfc/helpers/nfc_debug_pcap.c [hunk 25]
```diff
@@ -77,8 +77,8 @@ static void
         .event = event,
         .len = len << 8 | len >> 8,
     };
-    xStreamBufferSend(instance->stream, &pkt_hdr, sizeof(pkt_hdr), osWaitForever);
-    xStreamBufferSend(instance->stream, data, len, osWaitForever);
+    xStreamBufferSend(instance->stream, &pkt_hdr, sizeof(pkt_hdr), FuriWaitForever);
+    xStreamBufferSend(instance->stream, data, len, FuriWaitForever);
```
Comment (hedger) [inline applications/nfc/helpers/nfc_debug_pcap.c:81]: xStreamBufferSend

File: applications/nfc/nfc_worker.c [hunk 26]
```diff
@@ -434,7 +434,7 @@ void nfc_worker_mifare_classic_dict_attack(NfcWorker* nfc_worker) {
                     }
                 }
                 if(nfc_worker->state != NfcWorkerStateReadMifareClassic) break;
-                osDelay(1);
+                furi_delay_ms(1);
```
Comment (hedger) [inline applications/nfc/nfc_worker.c]: delay_ticks

File: applications/snake_game/snake_game.c [hunk 27]
```diff
@@ -299,21 +299,21 @@ int32_t snake_game_app(void* p) {
     view_port_draw_callback_set(view_port, snake_game_render_callback, &state_mutex);
     view_port_input_callback_set(view_port, snake_game_input_callback, event_queue);
 
-    osTimerId_t timer =
-        osTimerNew(snake_game_update_timer_callback, osTimerPeriodic, event_queue, NULL);
-    osTimerStart(timer, osKernelGetTickFreq() / 4);
+    FuriTimer* timer =
+        furi_timer_alloc(snake_game_update_timer_callback, FuriTimerTypePeriodic, event_queue);
+    furi_timer_start(timer, furi_kernel_get_tick_frequency() / 4);
```
Comment (hedger) [inline applications/snake_game/snake_game.c:304]: use ticks/ms conversion

File: applications/subghz/helpers/subghz_frequency_analyzer_worker.c [hunk 28]
```diff
@@ -100,7 +100,7 @@ static int32_t subghz_frequency_analyzer_worker_thread(void* context) {
     furi_hal_subghz_set_path(FuriHalSubGhzPathIsolate);
 
     while(instance->worker_running) {
-        osDelay(10);
+        furi_delay_ms(10);
```
Comment (hedger) [inline applications/subghz/helpers/subghz_frequency_analyzer_worker.c:103]: delay_ticks + fix delay

File: applications/subghz/helpers/subghz_frequency_analyzer_worker.c [hunk 29]
```diff
@@ -130,7 +130,7 @@ static int32_t subghz_frequency_analyzer_worker_thread(void* context) {
                 furi_hal_spi_release(&furi_hal_spi_bus_handle_subghz);
 
                 // delay will be in range between 1 and 2ms
-                osDelay(3);
+                furi_delay_ms(3);
```
Comment (hedger) [inline applications/subghz/helpers/subghz_frequency_analyzer_worker.c]: delay_ticks + fix delay

File: applications/subghz/helpers/subghz_frequency_analyzer_worker.c [hunk 30]
```diff
@@ -176,7 +176,7 @@ static int32_t subghz_frequency_analyzer_worker_thread(void* context) {
                     furi_hal_spi_release(&furi_hal_spi_bus_handle_subghz);
 
                     // delay will be in range between 1 and 2ms
-                    osDelay(3);
+                    furi_delay_ms(3);
```
Comment (hedger) [inline applications/subghz/helpers/subghz_frequency_analyzer_worker.c]: delay_ticks + fix delay

File: applications/subghz/views/receiver.c [hunk 31]
```diff
@@ -266,7 +266,7 @@ bool subghz_view_receiver_input(InputEvent* event, void* context) {
                 return true;
             });
         if(subghz_receiver->lock_count == 0) {
-            osTimerStart(subghz_receiver->timer, pdMS_TO_TICKS(1000));
+            furi_timer_start(subghz_receiver->timer, pdMS_TO_TICKS(1000));
```
Comment (hedger) [inline applications/subghz/views/receiver.c:269]: use furi ms/ticks conversion

File: applications/subghz/views/receiver.c [hunk 32]
```diff
@@ -280,7 +280,7 @@ bool subghz_view_receiver_input(InputEvent* event, void* context) {
                     return true;
                 });
             //subghz_receiver->lock = SubGhzLockOff;
-            osTimerStart(subghz_receiver->timer, pdMS_TO_TICKS(650));
+            furi_timer_start(subghz_receiver->timer, pdMS_TO_TICKS(650));
```
Comment (hedger) [inline applications/subghz/views/receiver.c:283]: use furi ms/ticks conversion

File: applications/subghz/views/subghz_test_carrier.c [hunk 33]
```diff
@@ -154,14 +154,14 @@ void subghz_test_carrier_enter(void* context) {
 
     furi_hal_subghz_rx();
 
-    osTimerStart(subghz_test_carrier->timer, osKernelGetTickFreq() / 4);
+    furi_timer_start(subghz_test_carrier->timer, furi_kernel_get_tick_frequency() / 4);
```
Comment (hedger) [inline applications/subghz/views/subghz_test_carrier.c:157]: use ms/ticks conversion?

File: applications/subghz/views/subghz_test_packet.c [hunk 34]
```diff
@@ -206,22 +206,22 @@ void subghz_test_packet_enter(void* context) {
 
     furi_hal_subghz_start_async_rx(subghz_test_packet_rx_callback, instance);
 
-    osTimerStart(instance->timer, osKernelGetTickFreq() / 4);
+    furi_timer_start(instance->timer, furi_kernel_get_tick_frequency() / 4);
```
Comment (hedger) [inline applications/subghz/views/subghz_test_packet.c:209]: use ms/tick conversion?

File: furi/core/semaphore.h [hunk 35]
```diff
@@ -0,0 +1,58 @@
...
+ *
+ * @return     The furi status.
+ */
+FuriStatus furi_semaphore_release(FuriSemaphore* instance);
+
+/** Get semaphore count
+ *
+ * @param      instance  The pointer to FuriSemaphore instance
+ *
+ * @return     { description_of_the_return_value }
```
Comment (hedger) [inline furi/core/semaphore.h]: ?

File: furi/core/event_flag.h [hunk 36]
```diff
@@ -0,0 +1,70 @@
...
+ * @param      instance  pointer to FuriEventFlag
+ */
+void furi_event_flag_free(FuriEventFlag* instance);
+
+/** Set flags
+ *
+ * @param      instance  pointer to FuriEventFlag
+ * @param[in]  flags     The flags
+ *
+ * @return     { description_of_the_return_value }
```
Comment (hedger) [inline furi/core/event_flag.h]: fill in

File: furi/core/event_flag.h [hunk 37]
```diff
@@ -0,0 +1,70 @@
...
+ */
+uint32_t furi_event_flag_set(FuriEventFlag* instance, uint32_t flags);
+
+/** Clear flags
+ *
+ * @param      instance  pointer to FuriEventFlag
+ * @param[in]  flags     The flags
+ *
+ * @return     { description_of_the_return_value }
```
Comment (hedger) [inline furi/core/event_flag.h]: same

File: furi/core/event_flag.h [hunk 38]
```diff
@@ -0,0 +1,70 @@
...
+ */
+uint32_t furi_event_flag_clear(FuriEventFlag* instance, uint32_t flags);
+
+/** Get flags
+ *
+ * @param      instance  pointer to FuriEventFlag
+ *
+ * @return     { description_of_the_return_value }
```
Comment (hedger) [inline furi/core/event_flag.h]: + rest in this file

File: applications/music_player/music_player_worker.c (hunk 3 above)
Comment (skotopes) [reply applications/music_player/music_player_worker.c:84]: Was meant to be 2 ms delay

File: furi/core/kernel.h [hunk 39]
```diff
@@ -0,0 +1,99 @@
...
+ *
+ * @param[in]  ticks  The tick until which kerel should delay task execution
+ *
+ * @return     The furi status.
+ */
+FuriStatus furi_delay_until_tick(uint32_t tick);
+
+/** Get instructions per microsecond count
+ *
+ * @return     { description_of_the_return_value }
```
Comment (hedger) [inline furi/core/kernel.h]: fill in

**Review by hedger — CHANGES_REQUESTED** [review]

File: applications/gui/modules/text_input.c (hunk 20 above)
Comment (skotopes) [reply applications/gui/modules/text_input.c:313]: in part 3

File: applications/input/input.c (hunk 21 above)
Comment (skotopes) [reply applications/input/input.c:9]: in part 3

File: applications/lfrfid/helpers/rfid_reader.cpp (hunk 23 above)
Comment (skotopes) [reply applications/lfrfid/helpers/rfid_reader.cpp:42]: Question for @DrZlo13

File: applications/music_player/music_player_worker.c (hunk 24 above)
Comment (skotopes) [reply applications/music_player/music_player_worker.c:54]: nope, ms

File: applications/nfc/helpers/nfc_debug_pcap.c (hunk 25 above)
Comment (skotopes) [reply applications/nfc/helpers/nfc_debug_pcap.c:81]: in part 3

File: applications/snake_game/snake_game.c (hunk 27 above)
Comment (skotopes) [reply applications/snake_game/snake_game.c:304]: in part 3

File: applications/subghz/helpers/subghz_frequency_analyzer_worker.c (hunk 28 above)
Comment (skotopes) [reply applications/subghz/helpers/subghz_frequency_analyzer_worker.c:103]: ms is ok here

File: applications/subghz/helpers/subghz_frequency_analyzer_worker.c (hunk 29 above)
Comment (skotopes) [reply applications/subghz/helpers/subghz_frequency_analyzer_worker.c]: nope ms here is intentional

File: applications/subghz/helpers/subghz_frequency_analyzer_worker.c (hunk 30 above)
Comment (skotopes) [reply applications/subghz/helpers/subghz_frequency_analyzer_worker.c]: same

File: applications/subghz/views/receiver.c (hunk 31 above)
Comment (skotopes) [reply applications/subghz/views/receiver.c:269]: in part 3

File: applications/subghz/views/receiver.c (hunk 32 above)
Comment (skotopes) [reply applications/subghz/views/receiver.c:283]: in part 3

File: applications/subghz/views/subghz_test_carrier.c (hunk 33 above)
Comment (skotopes) [reply applications/subghz/views/subghz_test_carrier.c:157]: in part 3

File: applications/subghz/views/subghz_test_packet.c (hunk 34 above)
Comment (skotopes) [reply applications/subghz/views/subghz_test_packet.c:209]: in part 3

File: applications/desktop/animations/animation_manager.c (hunk 14 above)
Comment (skotopes) [reply applications/desktop/animations/animation_manager.c:201]: In part 3

File: applications/desktop/animations/animation_manager.c (hunk 15 above)
Comment (skotopes) [reply applications/desktop/animations/animation_manager.c:514]: In part 3

File: applications/desktop/animations/animation_manager.c (hunk 16 above)
Comment (skotopes) [reply applications/desktop/animations/animation_manager.c:509]: In part 3

File: applications/gui/modules/popup.c (hunk 19 above)
Comment (skotopes) [reply applications/gui/modules/popup.c:96]: In part 3

**Review by DrZlo13 — APPROVED** [review]

**Comment by DrZlo13** [comment]
200000 files are changed, with a million more well on the way

**Review by hedger — APPROVED** [review]

**Pull request merged**